### PR TITLE
Implemented bid price adjustment

### PR DESCRIPTION
**Pull request by anestere**

### CODE REVIEW

File: src/main/java/org/prebid/server/auction/ExchangeService.java [hunk 1]
```diff
@@ -392,11 +401,14 @@ private static boolean shouldCacheBids(ExtRequestTargeting targeting, ExtBidRequ
...
     private Future<BidderResponse> requestBids(BidderRequest bidderRequest, long startTime, Timeout timeout,
-                                               Map<String, String> aliases) {
+                                               Map<String, String> aliases,
+                                               Map<String, BigDecimal> bidAdjustments) {
         final String bidder = bidderRequest.getBidder();
+        final BigDecimal bidPriceAdjustmentFactor = bidAdjustments.get(bidder);
         return bidderCatalog.bidderRequesterByName(resolveBidder(bidder, aliases))
                 .requestBids(bidderRequest.getBidRequest(), timeout)
                 .map(this::validateAndUpdateResponse)
+                .map(seat -> ExchangeService.applyBidPriceAdjustment(seat, bidPriceAdjustmentFactor))
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/auction/ExchangeService.java]: No need for full qualifier
```
.map(seat -> applyBidPriceAdjustment(seat, bidPriceAdjustmentFactor))
```

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 2]
```diff
@@ -82,10 +84,17 @@ public ValidationResult validate(BidRequest bidRequest) {
             final ExtBidRequest extBidRequest = parseAndValidateExtBidRequest(bidRequest);
 
             final ExtRequestPrebid extRequestPrebid = extBidRequest != null ? extBidRequest.getPrebid() : null;
-            Map<String, String> aliases = extRequestPrebid != null ? extRequestPrebid.getAliases() : null;
-            aliases = aliases != null ? aliases : Collections.emptyMap();
 
-            validateAliases(aliases);
+            final boolean isPrebidExtPresent = extRequestPrebid != null;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/validation/RequestValidator.java]: Do we need separate var for one time using?

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 3]
```diff
@@ -110,6 +119,31 @@ public ValidationResult validate(BidRequest bidRequest) {
         return ValidationResult.success();
     }
 
+    private void validateBidAdjustmentFactors(Map<String, BigDecimal> adjustmentFactors, Map<String, String> aliases)
+            throws ValidationException {
+
+        for (Map.Entry<String, BigDecimal> bidderAdjustment : adjustmentFactors.entrySet()) {
+            final String bidder = bidderAdjustment.getKey();
+            final BigDecimal adjustmentFactor = bidderAdjustment.getValue();
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/validation/RequestValidator.java]: Minor. We can move down adjustmentFactor before `if(...)` statement.

File: src/test/java/org/prebid/server/auction/ExchangeServiceTest.java [hunk 4]
```diff
@@ -1012,23 +1012,63 @@ public void shouldIncrementNoBidRequestsMetric() {
...
+        final BidResponse bidResponse = exchangeService.holdAuction(bidRequest, uidsCookie, timeout).result();
+
+        // then
+        assertThat(bidResponse.getSeatbid())
+                .flatExtracting(SeatBid::getBid)
+                .extracting(Bid::getPrice).containsExactly(BigDecimal.valueOf(2.0));
+    }
+
+    @Test
+    public void shouldRequestBidsWithBidAdjustmentFactor() {
```
Comment (rpanchyk) [inline src/test/java/org/prebid/server/auction/ExchangeServiceTest.java]: Duplicates `shouldReturnBidsWithAdjustedPricesWhenAdjustmentFactorPresent` test.

File: src/test/java/org/prebid/server/validation/RequestValidatorTest.java [hunk 5]
```diff
@@ -1497,6 +1498,70 @@ public void validateShouldReturnValidationMessageWhenMetricValueIsNotValid() {
...
+        final BidRequest bidRequest = validBidRequestBuilder()
+                .ext(mapper.valueToTree(ExtBidRequest.of(
+                        ExtRequestPrebid.of(null, singletonMap("rubicon", BigDecimal.valueOf(1.1)), null, null, null))))
+                .build();
+
+        // when
+        final ValidationResult result = requestValidator.validate(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(0);
```
Comment (rpanchyk) [inline src/test/java/org/prebid/server/validation/RequestValidatorTest.java]: `hasSize(0)` can be replaced with `isEmpty()`. Check another places pls.

**Comment by bretg** [comment]
@jordillachmrf also took a look at these changes and is ok with them.

**Pull request merged**